.gitlab-ci.yml

```diff
@@ -30,50 +126,45 @@ Test:
   - dev
   - master
   - /^feature\/.*$/
+  - /^migration\/.*$/
 
+# Job to build static asset for staging and live environment
+# dev branch -> Staging
+# master branch -> Live
 Test_N_Build:
-  image: node:12.13.0
-  stage: test_build_static
+  extends: .build-static
 
-  before_script:
-  - cd src/ensembl
-  - npm ci
+  variables:
+    GOOGLE_ANALYTICS_KEY: ${GA_KEY}
+    ENVIRONMENT: production
+    API_HOST: ""
 
   script:
   - npm run test
   - export NODE_ENV=production
-  - export GOOGLE_ANALYTICS_KEY=${GA_KEY}
-  - export ENVIRONMENT=production
-  - export API_HOST=""
   - npm run build
 
-  artifacts:
-    name: static_assets
-    paths:
-    - src/ensembl/dist/
-
   only:
   - dev
   - master
 
+# Job to build static asset for internal environment
+# dev branch -> Internal
 Test_N_Build:internal:
-  image: node:12.13.0
-  stage: test_build_static
+  extends: .build-static
 
-  before_script:
-  - cd src/ensembl
-  - npm ci
+  variables:
+    GOOGLE_ANALYTICS_KEY: ${GA_KEY}
+    ENVIRONMENT: internal
+    API_HOST: ""
 
   script:
   - npm run test
   - export NODE_ENV=development
```

Comment:
This should be production; but since this job extends `.build-static`, do we even need the `script` block?

Reply:
If NODE_ENV is production while building for internal deployment then yes we don't need script block in Test_N_Build:internal job.